### ISSUE TITLE
Support escaping of table characters inside a table

### ISSUE THREAD

**Issue by Wandmalfarbe** [MacDownApp/macdown#623]
A normal pipe table produces the desired table in MacDown.
```
test | a | table
--- | --- | --- 
one | two | three
```

A pipe table with an escaped pipe symbol in the header is not recognized as a table.

```
test | P(a\|b) | table
--- | --- | --- 
one | two | three
```
Please take a look at the [Babelmark output](http://johnmacfarlane.net/babelmark2/?normalize=1&text=test+%7C+a+%7C+table%0A---+%7C+---+%7C+---+%0Aone+%7C+two+%7C+three%0A%0Atest+%7C+p(a%5C%7Cb)+%7C+table%0A---+%7C+---+%7C+---+%0Aone+%7C+two+%7C+three) for this example. Desired output is currently produced by:

```
pandoc 1.17.1
PHP Markdown Extra 1.2.8
Minima 0.8.0a3_20140907
kramdown 1.2.0
cebe/markdown GFM 1.1.0
cebe/markdown MarkdownExtra 1.1.0
Blackfriday
Earmark 0.2.1
```

**Comment by AndrewFreemantle** [comment]
Also affects the table layout when the pipe is in a code block too. Escaping doesn't affect the output..

```
test | a    | table
---- |----- | -----
one  | two  | `t | h | r | e | e`
four | five | six
```

renders as:

test | a    | table
---- |----- | -----
one  | two  | \`t
four | five | six

**Comment by Jmuccigr** [comment]
Hoedown bug?

**Comment by franklinyu** [comment]
> Hoedown bug?

Yes, I think so. Currently, Hoedown with table (`hoedown --tables`) converts

```markdown
test | P(a\|b) | table
--- | --- | --- 
one | two | three
```

into (unexpected)

```html
<p>test | P(a|b) | table
--- | --- | --- 
one | two | three</p>
```

I have no idea how to render a table with develop branch; will file an issue to them.

**Comment by slimandslam** [comment]
+1 here.  My MD table renders fine on Github with this header:

```|      **Subclass \| Description**          |         **Method**            |    **Parameters**    |```      

But MacDown does not render it unless I get rid of the escaped '|' character
